### ISSUE TITLE
Possible typo MainLogging.defaultScreen Terminal.get usage?

### ISSUE THREAD

**Issue by er1c** [sbt/sbt#6678]
In `MainLogging`

https://github.com/sbt/sbt/blob/fe5497938a8b2898dcbab49b60e0f2ef53b06c29/internal/util-logging/src/main/scala/sbt/internal/util/MainLogging.scala#L69-L88

Line 78 uses `Terminal.get` based upon the other signatures, they use the passed in `console`

As far as I can tell the consoleOpt usage https://github.com/sbt/sbt/blob/fe5497938a8b2898dcbab49b60e0f2ef53b06c29/internal/util-logging/src/main/scala/sbt/internal/util/MainLogging.scala#L60 get ignored

**Comment by eed3si9n** [comment]
I am actually not sure if this was a typo.

Looking at git blame, this was changed in https://github.com/sbt/sbt/pull/5620/commits/1b03c9b1a9e65d174f2470486a6d2178ceb770fd to enable sbtn terminal forwarding. If we do change this area, we probably need to make sure sbtn continues to work.

**Comment by er1c** [comment]
@eatkins I can't tell from the change if this was actually intentional or not

**Comment by eatkins** [comment]
I don't remember writing this code all that well but I do remember that I did a bunch of refactoring. It is quite possible that in refactoring I failed to update an api unintentionally. If you think that it looks like a typo then there is a good chance that it is. To know for sure I'd have to reread all of the surrounding code. In general though I think `Terminal.get` is to be avoided, especially if the parameter list includes a parameter that provides a Terminal. My guess is that https://github.com/er1c/sbt/commit/c1ed109f8927cf8b1815c50ac45444cfd7cc76d3 is both safe and more correct but I'm not 100% sure.
